cylc/flow/graph_parser.py

```diff
@@ -469,10 +470,11 @@ def parse_graph(self, graph_string: str) -> None:
                 pairs.add((chain[i], chain[i + 1]))
 
         # Get a set of RH nodes which are not at the LH of another pair:
-        pairs_dict = dict(pairs)
+        pairs_dict = dict([p for p in pairs if p[0] is not None])
```

Comment:
I don't think that this was actually causing problems - but I'm concerned that this is bad code because it's very possible that `[i[0] for i in pairs]` may have repeat values, and I'm not sure what the consequences of that are.